ivy/functional/frontends/torch/nn/functional/loss_functions.py

```diff
@@ -315,13 +315,13 @@ def gaussian_nll_loss(
 
     if var.shape != input.shape:
         if input.shape[:-1] == var.shape:
-            var.unsqueeze(dim=2)
-    elif input.shape[:-1] == var.shape[:-1] and var.shape[-1] == 1:
-        pass
-    else:
-        ivy.utils.exceptions.IvyError(
-            f"var is of incorrect size"
-        )
+            var = torch_frontend.unsqueeze(var, dim=2)
+        elif input.shape[:-1] == var.shape[:-1] and var.shape[-1] == 1:
+            pass
+        else:
+            raise ivy.utils.exceptions.IvyError(
+                "var is of incorrect size"
+            )
```

Comment:
I still don't think these checks are consistent with what the documentation mentions, but this will become clearer during testing.